flake.nix

```diff
@@ -0,0 +1,181 @@
+        mlir-with-llvm = pkgs.symlinkJoin {
+          name = "mlir-with-llvm-${mlirVersion}";
+          paths = [
+            pkgs.llzk-llvmPackages.libllvm.dev
+            pkgs.llzk-llvmPackages.libllvm.lib
+            pkgs.llzk-llvmPackages.mlir.dev
+            pkgs.llzk-llvmPackages.mlir.lib
+          ];
+          postBuild = ''
+            echo "Creating merged package: $out"
+            mv "$out"/bin/llvm-config "$out"/bin/llvm-config-native
+            substitute ${./nix/llvm-config.sh.in} "$out"/bin/llvm-config \
+              --subst-var-by out "${placeholder "out"}" \
+              --subst-var-by cmakeBuildType "Release" \
+              --subst-var-by version "${mlirVersion}" 
+            chmod +x "$out"/bin/llvm-config
+            # Replace the MLIR dynamic library from the LLVM build with a dummy static library
+            # to avoid duplicate symbol issues when linking with both LLVM and MLIR since the
+            # MLIR build generated individual static libraries for each component.
+            rm -f "$out"/lib/libMLIR.${if pkgs.stdenv.isDarwin then "dylib" else "so"}
+            ${pkgs.stdenv.cc}/bin/ar -r "$out"/lib/libMLIR.a
+          '';
+        };
+
+        commonNativeBuildInputs = (
+          with pkgs;
+          [
+            cmake
+            rustc
+            cargo
+            clang
+          ]
+        );
+
+        commonBuildInputs = (
+          with pkgs;
+          [
+            libxml2
+            zlib
+            zstd
+            z3.lib
+            mlir-with-llvm
+          ]
+        );
+
+        # Helper function to build Rust packages with common configuration
+        buildLlzkRustPackage =
+          packageName:
+          pkgs.rustPlatform.buildRustPackage rec {
+            pname = "${packageName}-rs";
+            version = (pkgs.lib.importTOML ./${packageName}/Cargo.toml).package.version;
+            # Note: for this source to include the `llzk-lib` submodule, the nix command line
+            # must use `.?submodules=1`. For example, `nix build '.?submodules=1#llzk-rs'`.
+            src = ./.;
+
+            nativeBuildInputs = commonNativeBuildInputs;
+            buildInputs = commonBuildInputs;
+
+            cargoLock = {
+              lockFile = ./Cargo.lock;
+              allowBuiltinFetchGit = true;
+            };
+
+            # Build only the specified package from the workspace
+            cargoBuildFlags = [
+              "--package"
+              packageName
+            ];
+            cargoTestFlags = [
+              "--package"
+              packageName
+            ];
+
+            CC = "clang";
+            CXX = "clang++";
+            MLIR_SYS_200_PREFIX = "${mlir-with-llvm}";
+            TABLEGEN_200_PREFIX = "${mlir-with-llvm}";
+            LIBCLANG_PATH = "${pkgs.llzk-llvmPackages.libclang.lib}/lib";
+            RUSTFLAGS = "-lLLVM -L ${mlir-with-llvm}/lib/ -lz3 -L ${pkgs.z3.lib}/lib";
```

Comment:
@0xddom I had to add these additional libs to link successfully. Should these instead be configured in the `llzk-sys` build setup?

Reply:
Check this build from before I added `-lLLVM` : https://github.com/Veridise/llzk-rs/actions/runs/18193302940/job/51793153263

Reply:
FYI `-lLLVM` will link the shared library, which you'll get if you call `llvm-config --link-shared --libs`

Reply:
I see, the build scripts are made under the assumption that linking is static. Do static libraries have a different path in nix or something?